### PR TITLE
Added the BeanConverter class

### LINKED ISSUE #80: Add BeanConverter

### PR DESCRIPTION
**Pull request by chrisrohr**
Fixes #80

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -97,6 +102,12 @@
             <optional>true</optional>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-beans</artifactId>
+            <version>${spring.version}</version>
```
Comment (sleberknight) [inline pom.xml:108]: This needs to be provided scope

File: pom.xml [hunk 2]
```diff
@@ -10,14 +10,19 @@
     <packaging>jar</packaging>
 
     <properties>
-        <slf4j-api.version>1.7.30</slf4j-api.version>
-        <guava.version>28.2-jre</guava.version>
+        <!-- Dependency versions -->
```
Comment (sleberknight) [inline pom.xml:13]: Please revert these changes (I have a local branch with a bunch of new dependencies) and only add the spring.version property and the spring-beans dep

File: src/main/java/org/kiwiproject/beans/BeanConverter.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+import java.beans.PropertyDescriptor;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+@Slf4j
+public class BeanConverter<T> {
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/beans/BeanConverter.java:24]: Can you please add a top-level doc explaining what this is and why use it?

File: src/main/java/org/kiwiproject/beans/BeanConverter.java [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+    @Getter
+    private Map<String, Function<T, ?>> mappers = new HashMap<>();
```
Comment (sleberknight) [inline src/main/java/org/kiwiproject/beans/BeanConverter.java:27]: Can you please add docs explaining `mappers`, `exclusionList`, and `failOnError`?

File: pom.xml (hunk 1 above)
Comment (chrisrohr) [reply pom.xml]: yes, I did that in the revert

File: pom.xml [hunk 5]
```diff
@@ -97,6 +97,14 @@
             <optional>true</optional>
         </dependency>
 
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-beans</artifactId>
+            <version>5.2.5.RELEASE</version>
+            <scope>provided</scope>
+            <optional>true</optional>
```
Comment (sleberknight) [inline pom.xml:105]: I _think_ we should only use the provided scope, not optional. See https://medium.com/@danismaz.furkan/difference-between-optional-true-optional-and-scope-provided-scope-7404ec24fb59

File: src/main/java/org/kiwiproject/beans/BeanConverter.java (hunk 3 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/beans/BeanConverter.java:32]: Done

File: pom.xml (hunk 5 above)
Comment (chrisrohr) [reply pom.xml]: Removed

File: src/main/java/org/kiwiproject/beans/BeanConverter.java (hunk 4 above)
Comment (chrisrohr) [reply src/main/java/org/kiwiproject/beans/BeanConverter.java:38]: Done
